nvm.sh

```diff
@@ -1725,9 +1725,23 @@ nvm_download_artifact() {
     TARBALL_URL="${MIRROR}/${SLUG}.tar.${COMPRESSION}"
   fi
 
-  if nvm_compare_checksum "${TARBALL}" "${CHECKSUM}" >/dev/null 2>&1; then
-    nvm_err "Checksums match! Using existing downloaded archive $(nvm_sanitize_path "${TARBALL}")"
-  else
+  local USE_LOCAL_TARBALL
+  if [ -r "${TARBALL}" ]; then
+    nvm_err "Local cache found:"
+    nvm_err "$(nvm_sanitize_path "${TARBALL}")"
+    if nvm_compare_checksum "${TARBALL}" "${CHECKSUM}" >/dev/null 2>&1; then
+      nvm_err "Checksums match! Using existing downloaded archive $(nvm_sanitize_path "${TARBALL}")"
+      USE_LOCAL_TARBALL=true
+      nvm_echo "${TARBALL}"
+      return 0
+    fi
+    nvm_compare_checksum "${TARBALL}" "${CHECKSUM}"
+    nvm_err "Checksum check failed!"
+    nvm_err "Removing the broken local cache..."
+    command rm -rf "${TARBALL}"
+    nvm_err "Downloading ${TARBALL_URL} ..."
+  fi
+  if [ "${USE_LOCAL_TARBALL-}" != true ]; then
```

Comment:
now that you've denested, we no longer need this variable at all - please remove it, and denest lines 1745-1761.